Sources/StreamChatUI/ChatThreadList/ChatThreadListVC.swift

```diff
@@ -0,0 +1,160 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// The view controller used to display the list of threads the current user is participating.
+@available(iOSApplicationExtension, unavailable)
+open class ChatThreadListVC:
+    _ViewController,
+    ThemeProvider,
+    ChatThreadListControllerDelegate,
+    UITableViewDelegate,
+    UITableViewDataSource
+{
+    /// The `ChatThreadListController` instance that provides the threads data.
+    public var threadListController: ChatThreadListController
+
+    public init(threadListController: ChatThreadListController) {
+        self.threadListController = threadListController
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @available(*, unavailable)
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    /// The current thread list data.
+    public private(set) var threads: [ChatThread] = []
+
+    /// A boolean value that controls whether it is loading more threads at the moment or not.
+    private var isPaginatingThreads = false
+
+    /// A component responsible to handle when to load new channels.
+    private lazy var viewPaginationHandler: ViewPaginationHandling = {
+        ScrollViewPaginationHandler(scrollView: tableView)
+    }()
+
+    /// The `UITableView` instance that displays the thread list.
+    open private(set) lazy var tableView: UITableView = UITableView()
+        .withoutAutoresizingMaskConstraints
+        .withAccessibilityIdentifier(identifier: "tableView")
+
+    /// A router object responsible for handling navigation actions of this view controller.
+    open lazy var router: ChatThreadListRouter = components
+        .threadListRouter
+        .init(rootViewController: self)
+
+    override open func setUp() {
+        super.setUp()
+
+        threadListController.delegate = self
+        threadListController.synchronize()
+
+        tableView.register(ChatThreadListItemCell.self)
+        tableView.delegate = self
+        tableView.dataSource = self
+
+        viewPaginationHandler.bottomThreshold = 800
```

Comment:
what's this value?

Reply:
It is the prefetch threshold, it is like at what height of the list it should trigger a new page